lib/db2.js

```diff
@@ -96,10 +110,12 @@ DB2.prototype.update = function(model, where, data, options, cb) {
  * @param {Function} cb The callback function
  */
 DB2.prototype.destroyAll = function(model, where, options, cb) {
+  debug('DB2.prototype.destroyAll: model=%s, where=%j, options=%j',
+        model, where, options);
   var self = this;
   var stmt = self.buildDelete(model, where, options);
-  var id = self.idName(model);
-  var sql = 'SELECT \"' + id + '\" FROM OLD TABLE (' + stmt.sql + ')';
+  var idName = self.idName(model);
+  var sql = 'SELECT \"' + idName + '\" FROM OLD TABLE (' + stmt.sql + ')';
```

Comment:
@Amir-61 @jannyHou You have only answered part of that question.  It is entirely possible that the table in the database was created outside loopback and has an auto-generated column.  How does a user know what to set in the model properties to identify this as auto-generated such that it does not allow you to create if you pass an value in for that column?

Note this is well outside the scope of this issue.